### ISSUE TITLE
Map#setBearing handles floats incorrectly

### ISSUE THREAD

**Issue by flawyte** [mapbox/mapbox-gl-js#7735]
**mapbox-gl-js version**: 0.52.0

**browser**: Chrome 71.0.3578.98 (Official Build) (64-bit)

### Steps to Trigger Behavior

 1. Make sure the map's bearing is set to an integer value
 2. Set the bearing to a small float value
 3. Read the bearing ‒ it's not the right value

### Link to Demonstration

https://jsbin.com/fakoxufeve/edit?html,console

### Expected Behavior

Bearing should equal `0.1`.

### Actual Behavior

Bearing equals `0.10000000000002274`.

**Comment by flawyte** [comment]
The problem seems to occur even with integer values :
```js
map.setBearing(41);
map.getBearing(); // 40.99999999999999
```

**Comment by mourner** [comment]
I don't think we should set any guarantees on the bearing/center/zoom numbers being exactly equal since they go through certain math transformations, can change depending on map constraints, etc., so I'm inclined to close with "won't fix". What's your use case?

**Comment by flawyte** [comment]
I'm displaying a map with several UI inputs including a `<input type='number' step='1' />` for the rotation, so the user can use the up/down keys to change the map's parameters in a precise way without having to manually type the new value. I'm also listening to the map's `moveend` event to update the input's value if the user moves/rotates the map directly.

Here's what happens if the input's value is set to `40` and the user presses the up arrow key :
- the input's value increases to `41`
- `map.setBearing(41)` is called
- `moveend` is called back because the map changed, but `map.getBearing()` returns `40.99999999999999` instead of `41`
- the input is updated with `moveend`'s value and the user ends up seeing `40.99999999999999`

Of course I can just round the displayed value using `map.getBearing().toFixed()`, but imo it would be simpler and more consistent if the value wasn't changed by some mysterious internal processing, because whether I'm a user or a developer if it set the rotation to `41` I expect it to be `41`.

**Comment by CodingReaper1** [comment]
where is this setBearing even coming from

**Comment by CodingReaper1** [comment]
> The problem seems to occur even with integer values :
> 
> ```js
> map.setBearing(41);
> map.getBearing(); // 40.99999999999999
> ```

what even is setBearing where is it coming from
